WalletWasabi/Hwi/HwiProcessManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Hwi
+{
+	public static class HwiProcessManager
+	{
+		public static async Task EnsureHwiInstalledAsync(string dataDir)
+		{
+			var fullBaseDirectory = Path.GetFullPath(AppContext.BaseDirectory);
+			if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				if (!fullBaseDirectory.StartsWith('/'))
+				{
+					fullBaseDirectory.Insert(0, "/");
+				}
+			}
+
+			var hwiDir = Path.Combine(dataDir, "hwi");
+
+			string hwiPath;
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				hwiPath = $@"{hwiDir}\hwi.exe";
+			}
+			else // Linux or OSX
+			{
+				hwiPath = $@"{hwiDir}/hwi";
+			}
+
+			if (!File.Exists(hwiPath))
+			{
+				Logger.LogInfo($"HWI instance NOT found at {hwiPath}. Attempting to acquire it...", nameof(HwiProcessManager));
+				await InstallHwiAsync(fullBaseDirectory, hwiDir);
+			}
+			else if (new FileInfo(hwiPath).CreationTimeUtc < new DateTime(2019, 04, 13, 0, 0, 0, 0, DateTimeKind.Utc))
```

Comment:
Q: shouldn't we check by version with `./hwi --version`?

Reply:
I'd keep it here since this is what we're doing with Tor, too.